client/dashboard/components/settings/general/store-address.tsx

```diff
@@ -315,9 +315,7 @@ export function useGetCountryStateAutofill(
 }
 
 type StoreAddressProps = {
-	// Disable reason: The getInputProps type are not provided by the caller and source.
-	// eslint-disable-next-line @typescript-eslint/no-explicit-any
-	getInputProps: any;
+	getInputProps: ( key: string ) => React.HTMLProps< HTMLInputElement >;
```

Comment:
I believe these aren't a 1:1 match for HTML props.  `getInputProps` is being passed from the `Form` component.  Might be worth typing there and exporting so we can import here.

```ts
{
			value: string|boolean, // worth double checking this one, not sure if we can set to any value
			checked: boolean,
			selected: string,
			onChange: (value: string) => void,
			onBlur: () => void,
			className: string,
			help: string|null,
		}
```

https://github.com/woocommerce/woocommerce-admin/blob/dbd62e4f0671707833dceb26238d44b1f12d7b26/packages/components/src/form/index.js#L121-L133

Reply:
To be fair, while some of the values aren't interchangeable between HTML components, `help` is the only one that really needs to be typed.  So you could also combine props from `HTMLProps` and type that out.

Would also be beneficial to put that in the `Form` component (or at least a util if you don't have time to convert the entire file to TS) if we go that route.

Reply:
> I believe these aren't a 1:1 match for HTML props. getInputProps is being passed from the Form component. Might be worth typing there and exporting so we can import here.

Good catch!

> Would also be beneficial to put that in the Form component (or at least a util if you don't have time to convert the entire file to TS) if we go that route.

Thanks for the suggestion. 👍  I'll look into the `Form` component.

Reply:
I just updated the input prop type in 1c71931cace58857f096d3ec139d1d4928abed5d. It's a little complicated to convert the entire `From` component into TS. 

I put the type file in `client/utils`. But I think we should move it into `packages/components` later once we have output d.ts files. I think @rjchow is working on that.

Reply:
Definitely adds a lot of extra work to convert that component to TS; we can save that for a follow-up.  A comment about moving later would be awesome, but otherwise this looks great!